test/QueryRegistry.test.ts

```diff
@@ -102,6 +111,29 @@ describe('Query Registry Contract', () => {
                 'Deployment Id already registered'
             );
         });
+
+        it('cannot create a project with not authorised account in creatorRestrict mode', async () => {
+            const [metadata, version, deploymentId] = [metadatas[0], versions[0], deploymentIds[1]];
+            await expect(
+                queryRegistry.connect(wallet_1).createQueryProject(metadata, version, deploymentId)
+            ).to.be.revertedWith('Address is not authorised to operate query project');
+            expect(await queryRegistry.creatorWhitelist(wallet_1.address)).to.be.equal(false);
+            await queryRegistry.addCreator(wallet_1.address);
```

Comment:
Extract the `success` case to something like `authorised account can create project in creatorRestricted mode`